### PR TITLE
docs: edit strict-mode.md to remove references for opting in

### PR DESCRIPTION
**Pull request by kapunahelewong**
For @aikidave's consideration. 

@aikidave, do we need to rewrite `template-typecheck.md`?

### CODE REVIEW

File: aio/content/navigation.json [hunk 1]
```diff
@@ -776,11 +776,6 @@
                   "url": "guide/browser-support",
                   "title": "Browser Support",
                   "tooltip": "Browser support and polyfills guide."
-                },
-                {
-                  "url": "guide/strict-mode",
-                  "title": "Strict mode",
-                  "tooltip": "Reference documentation for Angular's strict mode."
```
Comment (gkalpak) [inline aio/content/navigation.json:783]: It doesn't make much of a difference here, but you could just hide it with `hidden: true`, but keep it in `navigation.json` to show a ToC.

**Comment by aikidave** [comment]
This looks good to me. It seems that we should, at the very least, remove the `strict mode` section in template-typecheck. However, I think an engineer needs to determine if we need to rework the entire topic, which may be more work than we anticipated.

@jelbourn, @IgorMinar: can you offer your thoughts on this?

**Comment by jelbourn** [comment]
I'm slightly confused by this change. The strict mode setting still _exists_, it's just enabled by default in v12+ in new applications. It seems like we still should have documentation that explains what this setting means, no?

**Comment by kapunahelewong** [comment]
> I'm slightly confused by this change. The strict mode setting still _exists_, it's just enabled by default in v12+ in new applications. It seems like we still should have documentation that explains what this setting means, no?

I've switched it to editing the doc and setting aside the archival. Is this more appropriate, @jelbourn?

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

File: aio/content/guide/template-typecheck.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 Just as TypeScript catches type errors in your code, Angular checks the expressions and bindings within the templates of your application and can report any type errors it finds.
 Angular currently has three modes of doing this, depending on the value of the `fullTemplateTypeCheck` and `strictTemplates` flags in the [TypeScript configuration file](guide/typescript-configuration).
 
+<!-- Are these modes still here with the default being strict mode? -->
```
Comment (jelbourn) [inline aio/content/guide/template-typecheck.md:8]: @kyliau or @mgechev would probably be able to clarify which options remain available in v12 - I'm not 100% sure myself

Comment (mgechev) [reply aio/content/guide/template-typecheck.md:8]: We haven't removed any options. Folks in strict mode can still use different template type checking.

Comment (kapunahelewong) [reply aio/content/guide/template-typecheck.md]: Thanks! Leaving section.

File: aio/content/guide/setup-local.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ To install Angular on your local system, you need the following:
 
 * **Node.js**
 
-  Angular requires an [active LTS or maintenance LTS](https://nodejs.org/about/releases) version of Node.js.
+  Angular requires a [current, active LTS, or maintenance LTS](https://nodejs.org/about/releases) version of Node.js.
```
Comment (jelbourn) [inline aio/content/guide/setup-local.md:34]: Is this a rebase error? @josephperrott just changed this terminology in #41820

File: aio/content/guide/strict-mode.md [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
 
 * Enables [`strict` mode in TypeScript](https://www.typescriptlang.org/tsconfig#strict), as well as other strictness flags recommended by the TypeScript team. Specifically, `forceConsistentCasingInFileNames`, `noImplicitReturns`,  `noFallthroughCasesInSwitch`.
 * Turns on strict Angular compiler flags [`strictTemplates`](guide/angular-compiler-options#stricttemplates), [`strictInjectionParameters`](guide/angular-compiler-options#strictinjectionparameters) and [`strictInputAccessModifiers`](guide/template-typecheck#troubleshooting-template-errors).
 * [Bundle size budgets](guide/build#configuring-size-budgets) have been reduced by ~75%.
 
 You can apply these settings at the workspace and project level.
 
-To create a new workspace and application using the strict mode, run the following command:
+Strict mode is the default for new CLI-generated projects.
+Using the basic `ng new` command to create a new workspace and application automatically uses strict mode, as in the following command:
```
Comment (jelbourn) [inline aio/content/guide/strict-mode.md:17]: This seems kind of repetitive: "Strict mode is the default", "automatically uses strict mode"

File: aio/content/guide/template-typecheck.md [hunk 5]
```diff
@@ -41,8 +41,13 @@ The following still have type `any`.
 * The `$event` object.
 * Safe navigation expressions.
 
+
 {@a strict-mode}
 
+<!-- Should we remove the strict mode section that follows? What about the rest of this document? Do the other, less strict, modes still apply?-->
+
+<!--
```
Comment (jelbourn) [inline aio/content/guide/template-typecheck.md:49]: I think that this section should stay, but the part about "Angular version 9" should be removed

File: aio/content/guide/setup-local.md (hunk 3 above)
Comment (kapunahelewong) [reply aio/content/guide/setup-local.md:34]: Oh yes, I just mixed that one up. Fixing!

**Review by jelbourn — APPROVED** [review]
LGTM

**Pull request closed without merging**